test/sql-luatest/view_test.lua

```diff
@@ -0,0 +1,467 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_each(function()
+    g.server = server:new({alias = 'master'})
+    g.server:start()
+    g.server:exec(function()
+        box.execute([[SET SESSION "sql_seq_scan" = true;]])
+    end)
+end)
+
+g.after_each(function()
+    g.server:drop()
+end)
+
+g.test_view = function()
+    g.server:exec(function()
+        -- Create space and view.
+        box.execute("CREATE TABLE t1(a INT, b INT, PRIMARY KEY(a, b));")
+        t.assert(box.space.t1 ~= nil)
+        box.execute("CREATE VIEW v1 AS SELECT a+b FROM t1;")
+        t.assert(box.space.v1 ~= nil)
+
+        -- View can't have any indexes.
+        local _, err = box.execute("CREATE INDEX i1 ON v1(a);")
+        local err1 = "Can't create or modify index 'i1' "
+        local err2 = "in space 'v1': views can not be indexed"
+        t.assert_equals(err.message, err1 .. err2)
+        local v1 = box.space.v1
+        _, err = pcall(function()
+            return v1:create_index('primary', {parts = {1, 'string'}})
+        end)
+        local error = "Can't modify space 'v1': can not add index on a view"
+        t.assert_equals(err.message, error)
```

Comment:
Let's use `assert_error_msg_equals` or `assert_error_msg_contains` in cases where `pcall` is used to catch the error.
For example, instead of
```Lua
        local v1 = box.space.v1
        _, err = pcall(function()
            return v1:create_index('primary', {parts = {1, 'string'}})
        end)
        local error = "Can't modify space 'v1': can not add index on a view"
        t.assert_equals(err.message, error)
```
you will have
```Lua
        local v1 = box.space.v1
        local opts = {parts = {1, 'string'}}
        exp_err = "Can't modify space 'v1': can not add index on a view"
        t.assert_error_msg_equals(exp_err, v1.create_index, v1, 'primary', opts)
```